IHP/PGListener.hs

```diff
@@ -231,17 +232,30 @@ notifyLoop listeningToVar listenToVar subscriptions = do
                             let inChan = get #inChan subscription
                             Queue.writeChan inChan notification
 
+    -- Initial delay (in microseconds)
+    let initialDelay = 500 * 1000
+    -- Max delay (in microseconds)
+    let maxDelay = 60 * 1000 * 1000
     -- This outer loop restarts the listeners if the database connection dies (e.g. due to a timeout)
-    forever do
-        result <- Exception.try innerLoop
-        case result of
-            Left (error :: SomeException) -> do
-                case fromException error of
-                    Just (error :: AsyncCancelled) -> throw error
-                    notification -> do
-                        let ?context = ?modelContext -- Log onto the modelContext logger
-                        Log.info ("PGListener is going to restart, loop failed with exception: " <> displayException error)
-            Right _ -> pure ()
+    let retryLoop delay = do
+            result <- Exception.try innerLoop
+            case result of
+                Left (error :: SomeException) -> do
+                    case fromException error of
+                        Just (error :: AsyncCancelled) -> throw error
+                        notification -> do
+                            let ?context = ?modelContext -- Log onto the modelContext logger
+                            Log.info ("PGListener is going to restart, loop failed with exception: " <> displayException error)
```

Comment:
Maybe worth indicating in the message the delay (i.e. the `nextDelay` value)?

Reply:
Yes sure! Does this look okay?